### ISSUE TITLE
GLSL externally-specified entry points

### ISSUE THREAD

**Issue by kvark** [gfx-rs/naga#72]
Current parser API requires the execution mode to be specified for the whole module:
```rust
naga::front::glsl::parse_str(&input, "main".to_string(), spirv::ExecutionModel::Vertex)
```
This isn't ideal, and this doesn't match other front-ends. Ideally, a GLSL file could be loaded containing multiple different entry points.

Here is what I suggest we do instead:
```rust
struct Options<'a> {
  entry_points: &'a FastHashMap<String, spirv::ExecutionModel>,
}
fn parse_str(source: &str, options: Options) -> crate::Module {..}
```

So basically we could externally specify the entry points, and support any number of them in a single GLSL file. Having the "external" data is required in some of the other end-points too: Metal backend needs to have the mapping of resources specified, for example.

**Comment by pjoe** [comment]
Not sure if still valid, but found: https://github.com/KhronosGroup/glslang/issues/605#issuecomment-264385414

**Comment by kvark** [comment]
Fixed by #447
